src/linker/Linker/UnconditionalSuppressMessageAttributeState.cs

```diff
@@ -0,0 +1,259 @@
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Mono.Cecil;
+
+namespace Mono.Linker
+{
+	public class UnconditionalSuppressMessageAttributeState
+	{
+		private readonly AssemblyDefinition _assembly;
+		private GlobalSuppressions _lazyGlobalSuppressions;
+		private readonly Dictionary<MetadataToken, ImmutableDictionary<string, SuppressMessageInfo>> _localSuppressionsByMdToken;
```

Comment:
Is there a reason we store these based on actual tokens? Why not use the Cecil OM - `IMetadataTokenProvider`? (Maybe the GetHashCode is too expensive - which it might be - in which case please add a comment describing why we do it this way)

Reply:
It made sense to me to use to most specific thing we care about when comparing members/types. I've changed this to use the md provider instead; I didn't see any time burden we should worry about.